ir/dump.cpp

```diff
@@ -33,11 +33,10 @@ using namespace P4::literals;
 namespace {
 class IRDumper : public Inspector {
     std::ostream &out;
-    std::set<const IR::Node *> dumped;
     unsigned maxdepth;
     cstring ignore;
     bool source;
-    bool preorder(const IR::Node *n) override {
+    bool firstLine(const IR::Node *n) {
```

Comment:
Unclear what is meant by firstLine and this function has side-effects. A comment could be helpful here.